util/network-devp2p/src/discovery.rs

```diff
@@ -518,7 +518,14 @@ impl<'a> Discovery<'a> {
 
 	fn on_ping(&mut self, rlp: &Rlp, node_id: &NodeId, from: &SocketAddr, echo_hash: &[u8]) -> Result<Option<TableUpdates>, Error> {
 		trace!(target: "discovery", "Got Ping from {:?}", &from);
-		let ping_from = NodeEndpoint::from_rlp(&rlp.at(1)?)?;
+		let ping_from = if let Ok(rlp) = rlp.at(1) {
+			NodeEndpoint::from_rlp(&rlp)?;
+		} else {
+			NodeEndpoint {
+				address: from.clone(),
+				udp_port: from.port()
```

Comment:
I'm not 100% sure this would be correct. The issue is that here the TCP port would be set as the UDP port, and the node will be added as a syncing node even though we can infer from the empty `ping_from` that it isn't.
We should have NodeEnpoint as: IP from `from`, UDP port from `from` and TCP port as `null` or `0`.
However, having this will fail the `endpoint.is_valid` method, because of the null TCP port. So we should also handle this case as a _bootnode only_ endpoint IMO